jetbrains-core/src/software/aws/toolkits/jetbrains/core/toolwindow/ToolkitToolWindow.kt

```diff
@@ -33,14 +33,15 @@ class ToolkitToolWindowManager(private val project: Project) {
     inner class ManagedToolkitToolWindow(private val type: ToolkitToolWindowType) : ToolkitToolWindow {
         private val tabs = mutableMapOf<String, ManagedToolkitToolWindowTab>()
 
-        override fun addTab(title: String, component: JComponent, activate: Boolean, id: String): ToolkitToolWindowTab {
+        override fun addTab(title: String, component: JComponent, activate: Boolean, id: String, disposable: Disposable?): ToolkitToolWindowTab {
             val content = ContentImpl(component, title, true)
             val toolWindow = windowManager.getToolWindow(type.id)
                 ?: windowManager.registerToolWindow(type.id, true, type.anchor, project, true).also {
                     it.setIcon(type.icon)
                     it.stripeTitle = type.title
                 }
             Disposer.register(content, Disposable { closeWindowIfEmpty(toolWindow, type.id) })
+            disposable?.let { Disposer.register(content, it) }
```

Comment:
This seems inverted, shouldnt we register when we close the tab (content) that we dispose? Not dispose the tab?

Reply:
content is the parent, that's how it already is?

Reply:
this is right, im a dummy